File: Strand/BLE/WhoopModel.swift
import CoreBluetooth

/// Which strap the user is pairing. The user picks this before scanning so we
/// look for exactly one device family instead of guessing — a WHOOP 4.0 scan no
/// longer waits forever on a WHOOP 5/MG wrist, and vice versa.
public enum WhoopModel: String, CaseIterable, Identifiable, Hashable {
    case whoop4   = "WHOOP 4.0"
    case whoop5mg = "WHOOP 5.0 / MG"

    public var id: String { rawValue }
    public var displayName: String { rawValue }

    /// The model the user last chose, read from the same key the pickers write
    /// (`@AppStorage("selectedWhoopModel")`). Used as the default for scans the user
    /// didn't directly trigger — BLE state restoration, power-on reconnect — so those
    /// look for the right strap after a relaunch instead of falling back to WHOOP 4.0.
    public static var persisted: WhoopModel {
        UserDefaults.standard.string(forKey: "selectedWhoopModel").flatMap(WhoopModel.init(rawValue:)) ?? .whoop4
    }

    /// The BLE service to scan for, and to discover after connecting, for this model.
    /// These mirror `BLEManager.customService` / `BLEManager.whoop5Service` (kept inline
    /// here so the enum stays nonisolated — `BLEManager` is `@MainActor`). CBUUID compares
    /// by value, so these match the manager's constants in every `switch`/scan filter.
    public var scanService: CBUUID {
        switch self {
        case .whoop4:   return CBUUID(string: "61080001-8d6d-82b8-614a-1c8cb0f8dcc6")
        case .whoop5mg: return CBUUID(string: "fd4b0001-cce1-4033-93ce-002d5875f58a")
        }
    }
}

File: android/app/src/main/java/com/noop/ble/WhoopModel.kt
package com.noop.ble

import java.util.UUID

/**
 * Which strap the user is pairing. They pick this before scanning so we look for
 * exactly one device family instead of guessing — a WHOOP 4.0 scan no longer
 * waits forever on a WHOOP 5/MG wrist, and vice versa.
 *
 * This is the user-facing choice; it is deliberately separate from the
 * protocol-layer DeviceFamily (which carries CRC/characteristic detail).
 */
enum class WhoopModel(val displayName: String, val service: UUID) {
    WHOOP4("WHOOP 4.0", WhoopBleClient.WHOOP4_SERVICE),
    WHOOP5_MG("WHOOP 5.0 / MG", WhoopBleClient.WHOOP5_SERVICE),
}
